pandas/core/arrays/period.py

```diff
@@ -85,94 +108,131 @@ def wrapper(self, other):
+        ndarray[int64]), or a box around such an array (Series[period],
+        PeriodIndex).
+    freq : str or DateOffset
+        The `freq` to use for the array. Mostly applicable when `values`
+        is an ndarray of integers, when `freq` is required. When `values`
+        is a PeriodArray (or box around), it's checked that ``values.freq``
+        matches `freq`.
+    copy : bool, default False
+        Whether to copy the ordinals before storing.
+
+    Notes
+    -----
+    There are two components to a PeriodArray
+
+    - ordinals : integer ndarray
+    - freq : pd.tseries.offsets.Offset
+
+    The values are physically stored as a 1-D ndarray of integers. These are
+    called "ordinals" and represent some kind of offset from a base.
+
+    The `freq` indicates the span covered by each element of the array.
+    All elements in the PeriodArray have the same `freq`.
+
+    See Also
+    --------
+    period_array : Create a new PeriodArray
+    pandas.PeriodIndex : Immutable Index for period data
+    """
+    _attributes = ["freq"]
+    _typ = "periodarray"  # ABCPeriodArray
+
+    # Names others delegate to us
+    _other_ops = []
+    _bool_ops = ['is_leap_year']
+    _object_ops = ['start_time', 'end_time', 'freq']
+    _field_ops = ['year', 'month', 'day', 'hour', 'minute', 'second',
+                  'weekofyear', 'weekday', 'week', 'dayofweek',
+                  'dayofyear', 'quarter', 'qyear',
+                  'days_in_month', 'daysinmonth']
+    _datetimelike_ops = _field_ops + _object_ops + _bool_ops
+    _datetimelike_methods = ['strftime', 'to_timestamp', 'asfreq']
 
     # --------------------------------------------------------------------
     # Constructors
+    def __init__(self, values, freq=None, copy=False):
+        if freq is not None:
+            freq = Period._maybe_convert_freq(freq)
 
-    _attributes = ["freq"]
+        if isinstance(values, ABCSeries):
+            values = values._values
+            if not isinstance(values, type(self)):
+                raise TypeError("Incorrect dtype")
 
-    def __new__(cls, values, freq=None, **kwargs):
-        if is_period_dtype(values):
-            # PeriodArray, PeriodIndex
-            if freq is not None and values.freq != freq:
-                raise IncompatibleFrequency(freq, values.freq)
-            freq = values.freq
-            values = values.asi8
+        elif isinstance(values, ABCPeriodIndex):
+            values = values._values
 
-        elif is_datetime64_dtype(values):
-            # TODO: what if it has tz?
-            values = dt64arr_to_periodarr(values, freq)
+        if isinstance(values, type(self)):
+            if freq is not None and freq != values.freq:
+                msg = DIFFERENT_FREQ_INDEX.format(values.freq.freqstr,
+                                                  freq.freqstr)
+                raise IncompatibleFrequency(msg)
+            values, freq = values._data, values.freq
```

Comment:
On the call we discussed avoiding `_data`.  Did that get un-done?

Reply:
IIRC, we talked about two things

1. Avoiding the name `._data` for the attribute storing the actual values (conflict with use in blocks, and elsewhere)
2. Standardize the name(s) used by the array

I've punted on 1 since I don't know what a better name would be, and that would require additional changes in DatetimeLikeArrayMixin, which uses `._data`, in addition to `.values` and `._ndarray_values`.

PeriodArray should just use `.asi8` for places it needs an integer array (one more push coming fixing a few ndarray_values I missed).

Reply:
> I've punted on 1 since I don't know what a better name would be

On the call we discussed setting `_ndarray_values` directly in `__init__`.  Don't worry about this too much; if I'm the only one with a strong opinion I can push to change it in a follow-on PR.